large_image/cache_util/cachefactory.py

```diff
@@ -89,32 +121,23 @@ def getCacheSize(self, numItems, cacheName=None):
         return numItems
 
     def getCache(self, numItems=None, cacheName=None):
+        loadCaches()
         # memcached is the fallback default, if available.
-        cacheBackend = config.getConfig('cache_backend', 'python')
+        cacheBackend = config.getConfig('cache_backend', None)
+
+        if cacheBackend is None and len(availableCaches):
+            cacheBackend = next(iter(availableCaches))
+            config.getConfig('logprint').info('Automatically setting `%s` as cache_backend from availableCaches' % cacheBackend)
+            config.setConfig('cache_backend', cacheBackend)
+
         if cacheBackend:
             cacheBackend = str(cacheBackend).lower()
+
         cache = None
-        if cacheBackend == 'memcached' and MemCache and numItems is None:
-            # lock needed because pylibmc(memcached client) is not threadsafe
-            cacheLock = threading.Lock()
+        # TODO: why have this numItems check?
```

Comment:
numItems is None for the tile caches and not None for the source cache.  The source cache has to use in-process caching (hence, python), since it includes open file handles.  There should probably be a clearer mechanism for this (e.g., some flag like "in-process") that would be used instead of this cryptic code.